### DISCUSSION TITLE
Custom Maps

### DISCUSSION THREAD

**Discussion by AhmadAmine998** [alexliniger/MPCC#73] [Q&A] [answered]
I wish to use this code to generate a trajectory for a custom racetrack of mine. Are there any guidelines I should follow when generating maps for the solver? I tried to use just a simple rectangular hallway, but splinify always failed due to duplicates. 

Edit: Below is the "track" that I am trying to run MPCC on
![track](https://user-images.githubusercontent.com/35192516/160474054-6171f3a8-8b34-4b7f-ac16-b47e1ef1f865.jpg)

**Comment by alexliniger** [comment, marked as answer]
The ideal centerline has a continuous changing curvature, but combination of straight lines and fixed radius curves does also work. 

Currently the code does only support boundaries which are uniform on the left and the right side (in fact the left right boundaries are only used for plotting). Changing this is possible, and I have the code, but not for MPCC, as I implemented it for my version of MPC-Curv which we used the Indy Autonomous Challenge.

**Reply by AhmadAmine998** [reply]
I see, thank you! Is your version available as an open-source repository as well?
